src/client/app/components/Mermaid.vue

```diff
@@ -0,0 +1,32 @@
+<script lang="ts" setup>
+import { ref, onMounted } from 'vue'
+import { useRoute } from '../router.js'
+import mermaid from 'mermaid'
+
+const { src } = defineProps<{
+  src: string
+}>()
+
+const svg = ref<string | null>(null)
+const route = useRoute()
+
+onMounted(() => {
+  mermaid.initialize({
+    startOnLoad: false,
+    theme: 'dark',
```

Comment:
This needs some change as currently the theme is dark on light mode too:

![image](https://user-images.githubusercontent.com/40380293/192519845-0ff1c734-98d2-46b5-8957-8468eb3c4f31.png)

Emerson is using a mutation observer for this: https://github.com/emersonbottero/vitepress-plugin-mermaid/blob/main/src/Mermaid.vue, but IG this can be much simplified if we somehow expose something like isDark.

Also, slidev has done some cool optimizations that we can include too: https://github.com/slidevjs/slidev/blob/main/packages/client/builtin/Mermaid.vue

I'll let you know once I manage to expose isDark (current logic resides in https://github.com/vuejs/vitepress/blob/main/src/client/theme-default/components/VPSwitchAppearance.vue).

Reply:
Thanks. I will check the above code and wait for the implementation to expose isDark.